bsp/boards/MSP430F5438A/spi.c

```diff
@@ -0,0 +1,236 @@
+/**
+\brief wsn430v13b-specific definition of the "spi" bsp module.
+
+\author Thomas Watteyne <watteyne@eecs.berkeley.edu>, February 2012.
+*/
+
+#include "msp430f5438a.h"
+#include "spi.h"
+#include "leds.h"
+#include "bsp.h"
+
+//=========================== defines =========================================
+
+//=========================== variables =======================================
+
+typedef struct {
+   // information about the current transaction
+   uint8_t*        pNextTxByte;
+   uint8_t         numTxedBytes;
+   uint8_t         txBytesLeft;
+   spi_return_t    returnType;
+   uint8_t*        pNextRxByte;
+   uint8_t         maxRxBytes;
+   spi_first_t     isFirst;
+   spi_last_t      isLast;
+   // state of the module
+   uint8_t         busy;
+#ifdef SPI_IN_INTERRUPT_MODE
+   // callback when module done
+   spi_cbt         callback;
+#endif
+} spi_vars_t;
+
+spi_vars_t spi_vars;
+
+//=========================== prototypes ======================================
+
+//=========================== public ==========================================
+
+void spi_init() {
+   // clear variables
+   memset(&spi_vars,0,sizeof(spi_vars_t));  
+   
+   UCB0CTL1   |=  UCSWRST ;                       // reset state
+   
+   // RF SPI0 CS as GPIO output high
+   //
+   P3SEL &= ~TRXEM_SPI_SC_N_PIN;
+   P3OUT |=  TRXEM_SPI_SC_N_PIN;
+   P3DIR |=  TRXEM_SPI_SC_N_PIN;
+   /* Configuration
```

Comment:
for multi-line comments, please use `//` rather than `/*`-style comments. This allows us to comment out entire sections with `/*` when debugging.